src/DssSpell.t.sol

```diff
@@ -495,5 +497,83 @@ contract DssSpellTest is DssSpellTestBase {
         // Validate post-spell state
         assertEq(arbitrumGateway.validDomains(arbDstDomain), 0, "l2-arbitrum-invalid-dst-domain");
     }
-}
 
+    function testFlapperUniV2() public {
+        // Create surplus manipulating dai and sin (ONLY FOR GOERLI)
+        vm.store(
+            address(vat),
+            keccak256(abi.encode(address(vow), uint256(5))),
+            bytes32(60_000_000 * RAD)
+        );
+        vm.store(
+            address(vat),
+            keccak256(abi.encode(address(vow), uint256(6))),
+            bytes32(5_000_000 * RAD)
+        );
+        vm.store(
+            address(vow),
+            bytes32(uint256(5)),
+            bytes32(3_000_000 * RAD)
+        );
+        vm.store(
+            address(vow),
+            bytes32(uint256(6)),
+            bytes32(2_000_000 * RAD)
+        );
+        //
+
+        address old_flap = chainLog.getAddress("MCD_FLAP");
+
+        assertEq(vow.flapper(), old_flap);
+        assertEq(vat.can(address(vow), old_flap),      1);
+        assertEq(vat.can(address(vow), address(flap)), 0);
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(vow.flapper(), address(flap));
+        assertEq(vat.can(address(vow), old_flap),      0);
+        assertEq(vat.can(address(vow), address(flap)), 1);
+
+        address pip = flap.pip();
+        assertEq(pip, addr.addr("PIP_MKR"));
+
+        address pair = flap.pair();
+
+        GemAbstract mkr = GemAbstract(addr.addr("MCD_GOV"));
+
+        // Generate liquidity in the pool
+        vm.store(
+            pip,
+            keccak256(abi.encode(address(this), uint256(4))),
+            bytes32(uint256(1))
+        );
+        uint256 price = MedianAbstract(pip).read();
+        uint256 daiAmt = 1_000_000 * WAD;
+        _giveTokens(address(dai), daiAmt);
+        dai.transfer(pair, daiAmt);
+        uint256 mkrAmt = 1_000_000 * WAD * WAD / price;
+        _giveTokens(address(mkr), mkrAmt);
+        mkr.transfer(pair, mkrAmt * 97 / 100); // 3% worse price (should fail)
```

Comment:
Should we maybe take into account any pre-existing MKR & DAI balance of the pair? I know they are both 0 here but on mainnet the pre-existing reserves could be such that we are not guaranteed that this transfer skews the pair price sufficiently far.

Reply:
Or alternatively I guess we could use a very large amount of test liquidity. I see that the liquidity of MKR/DAI in UniV2 is currently ~128$ so 1m$ of test liquidity should work fine but maybe we can boost that to e.g. 1bn$ to be on the safe side?

Reply:
Yeah I was actually thinking on that earlier today as well, I think I'll just overwrite the balances of the pair directly, wanted to make use of `_giveTokens` but just realized it was updated to a simple `GodMode.setBalance(token, address(this), amount);` so I can use the forge primitive directly in the pool.

Reply:
Changed!